src/main/scala/au/com/agiledigital/toolform/tasks/generate/docker/GenerateDockerComposeV3.scala

```diff
@@ -0,0 +1,175 @@
+package au.com.agiledigital.toolform.tasks.generate.docker
+
+import java.io.{BufferedWriter, FileWriter}
+
+import au.com.agiledigital.toolform.app.{ToolFormConfiguration, ToolFormError}
+import au.com.agiledigital.toolform.model._
+import au.com.agiledigital.toolform.tasks.generate.{WriterContext, YamlWriter}
+import au.com.agiledigital.toolform.tasks.generate.docker.DockerFormatting._
+import au.com.agiledigital.toolform.tasks.generate.docker.SubEdgeDef.subEdgeDefsFromProject
+import au.com.agiledigital.toolform.version.BuildInfo
+
+import scalaz.Scalaz._
+import scalaz._
+
+class GenerateDockerComposeV3() extends YamlWriter {
+
+  // TODO: Surely Scalaz has a built in identity function that could replace this.
+  // Maybe it could be replaced with a different construct altogether?
+  private val identity = State[WriterContext, Unit] { context =>
+    (context, ())
+  }
+
+  def runDockerComposeV3(toolFormConfiguration: ToolFormConfiguration, project: Project): Either[ToolFormError, String] = {
+    val outFile = toolFormConfiguration.generateTaskConfiguration.out
+    if (outFile.isDirectory) {
+      return Left(ToolFormError("Output path is a directory. Docker Compose V3 output requires a single file as an output."))
```

Comment:
This should be checked when parsing the options for the command. Kubernetes generation should parse options differently.

Reply:
I have removed this check from `GenerateDockerComposeV3`

I would add the validation to the command line parsing but I've heard that is changing soon anyway so I'll leave it for now.